server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -177,10 +199,82 @@ fn generate(
         .map(|line| line.item_id)
         .collect();
 
-    let requisition_line_rows =
+    let requisition_lines =
         generate_requisition_lines(ctx, &ctx.store_id, &requisition, program_item_ids)?;
 
-    Ok((requisition, requisition_line_rows))
+    let program_indicators = program_indicators(
+        connection,
+        Pagination::all(),
+        None,
+        Some(ProgramIndicatorFilter::new().program_id(EqualFilter::equal_to(&program.id))),
+    )?;
+
+    let indicator_values =
+        generate_internal_order_indicators(InternalOrderIndicatorGenerationInput {
+            store_id: ctx.store_id.clone(),
+            period_id,
+            program_indicators,
+            other_party_id,
+            connection,
+        })?;
+
+    Ok(GenerateResult {
+        requisition,
+        requisition_lines,
+        indicator_values,
+    })
+}
+
+pub struct InternalOrderIndicatorGenerationInput<'a> {
+    pub connection: &'a StorageConnection,
+    pub store_id: String,
+    pub period_id: String,
+    pub program_indicators: Vec<ProgramIndicator>,
+    pub other_party_id: String,
+}
+
+fn generate_internal_order_indicators(
+    input: InternalOrderIndicatorGenerationInput,
+) -> Result<Vec<IndicatorValueRow>, RepositoryError> {
+    let stores_with_store_as_supplier_ids: Vec<String> = NameRepository::new(input.connection)
+        .query(
+            &input.store_id,
+            Pagination::all(),
+            Some(NameFilter::new().supplying_store_id(EqualFilter::equal_to(&input.store_id))),
+            None,
+        )?
+        .into_iter()
+        .map(|s| s.store_row.and_then(|s| Some(s.id)))
+        .filter_map(|id| id)
+        .collect::<Vec<String>>();
+
+    if stores_with_store_as_supplier_ids.len() > 0 {
+        return Ok(generate_aggregate_indicator_values(
+            AggregateInternalOrderIndicatorGenerationInput {
+                store_id: input.store_id.clone(),
+                period_id: input.period_id,
+                program_indicators: input.program_indicators,
+                other_party_id: input.store_id.clone(),
+                customer_store_ids: stores_with_store_as_supplier_ids,
+                connection: input.connection,
+            },
+        )?);
+    } else {
+        return Ok(generate_program_indicator_values(
+            IndicatorGenerationInput {
+                store_id: input.store_id.clone(),
+                period_id: input.period_id,
+                program_indicators: input.program_indicators,
+                other_party_id: input.store_id,
```

Comment:
You're passing in the store_id but the indicator_value is saving the name_link_id. This will give a foreign constraint error..